sisl/geom/nanoribbon.py

```diff
@@ -0,0 +1,147 @@
+import numpy as np
+
+from sisl import geom, Atom, Geometry
+
+__all__ = ['nanoribbon', 'graphene_nanoribbon', 'agnr', 'zgnr']
+
+
+def nanoribbon(bond, atom, width, kind='armchair'):
+    r""" Construction of a nanoribbon unit cell of type armchair or zigzag.
+    The geometry is oriented along the x-axis.
+
+    Parameters
+    ----------
+    bond : float
+       bond length between atoms in the honeycomb lattice
+    atom : Atom
+       atom (or atoms) in the honeycomb lattice
+    width : int
+       number of atoms in the transverse direction
+    kind : {'armchair', 'zigzag'}
+       type of ribbon
+
+    See Also
+    --------
+    honeycomb : honeycomb lattices
+    graphene : graphene geometry
+    graphene_nanoribbon : graphene nanoribbon
+    agnr : armchair graphene nanoribbon
+    zgnr : zigzag graphene nanoribbon
+    """
+    if not isinstance(bond, float):
+        raise ValueError("nanoribbon: bond needs to be a float!")
+    if not isinstance(atom, (Atom, list, tuple)):
+        raise ValueError("nanoribbon: atom needs to be an instance of Atom (or list of Atoms)!")
+    if not isinstance(width, int):
```

Comment:
I will merge, but for your information, this is the correct way to do it:
```python
from numbers import Integral

... isinstance(width, Integral)
```
The problem is that `isinstance(np.int32(1), int) is False`, so it will fail if users pass an element from a numpy array.